src/ControlSystem/ControlErrors/Size/ComovingCharSpeedDerivative.cpp

```diff
@@ -0,0 +1,140 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "ControlSystem/ControlErrors/Size/ComovingCharSpeedDerivative.hpp"
+
+#include <cmath>
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tags/TempTensor.hpp"
+#include "DataStructures/TempBuffer.hpp"
+#include "DataStructures/Tensor/IndexType.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace control_system::size {
+template <typename Frame>
+void comoving_char_speed_derivative(
+    const gsl::not_null<Scalar<DataVector>*> result, const double lambda_00,
+    const double dt_lambda_00, const double horizon_00,
+    const double dt_horizon_00, const double grid_frame_excision_sphere_radius,
+    const tnsr::i<DataVector, 3, Frame>& excision_rhat,
+    const tnsr::i<DataVector, 3, Frame>& excision_normal_one_form,
+    const Scalar<DataVector>& excision_normal_one_form_norm,
+    const tnsr::I<DataVector, 3, Frame>& frame_components_of_grid_shift,
+    const tnsr::II<DataVector, 3, Frame>&
+        inverse_spatial_metric_on_excision_boundary,
+    const tnsr::Ijj<DataVector, 3, Frame>& spatial_christoffel_second_kind,
+    const tnsr::i<DataVector, 3, Frame>& deriv_lapse,
+    const tnsr::iJ<DataVector, 3, Frame>& deriv_shift) {
+  const double Y00 = 0.25 * M_2_SQRTPI;
+
+  // Define temporary storage.
+  using excision_normal_vector_tag = ::Tags::TempI<1, 3, Frame, DataVector>;
+  TempBuffer<tmpl::list<excision_normal_vector_tag>> buffer(
+      get<0>(excision_rhat).size());
+  auto& excision_normal_vector = get<excision_normal_vector_tag>(buffer);
+
+  for (size_t i = 0; i < 3; ++i) {
+    excision_normal_vector.get(i) =
+        get<0>(excision_normal_one_form) *
+            inverse_spatial_metric_on_excision_boundary.get(0, i) +
+        get<1>(excision_normal_one_form) *
+            inverse_spatial_metric_on_excision_boundary.get(1, i) +
+        get<2>(excision_normal_one_form) *
+            inverse_spatial_metric_on_excision_boundary.get(2, i);
+  }
```

Comment:
[optional] Consider
```c++

tenex::evaluate<ti::I>(
    make_not_null(&excision_normal_vector),
    excision_normal_one_form(ti::j) *
    inverse_spatial_metric_on_excision_boundary(ti::J, ti::I));
```
and perhaps similar things below.

Reply:
Added many tenex things. Cannot do everywhere because of frame differences (see comment).